packages/nimble-components/src/table/specs/table-column-specs/table-column-file-size.md

```diff
@@ -0,0 +1,222 @@
+        if (this.columnConfig) {
+            this.text = formatFileSize(
+                this.columnConfig.formatter, // instance of Intl.NumberFormat
+                this.unitStrings, // either ['byte', 'bytes', 'KB', ...] or ['byte', 'bytes', 'KiB', ...]
+                this.columnConfig.unitType,
+                this.cellRecord?.value
+            );
+        } else {
+            this.text = '';
+        }
+    }
+}
+
+// Shared function also used by group header view
+function formatFileSize(
+    formatter: Intl.NumberFormat,
+    unitStrings: string[],
+    unitType: FileSizeUnitType,
+    byteCount: number
+): string {
+    let currentSize = byteCount;
+    let unitIndex = 0;
+    if (currentSize !== 1) {
+        const divisor = unitType === FileSizeUnitType.binary ? 1024 : 1000;
+        while (currentSize >= divisor && unitIndex < unitStrings.length) {
+            currentSize /= divisor;
+            unitIndex += 1;
+        }
+    }
+
+    return `${formatter.format(currentSize)} ${unitStrings[unitIndex]}`;
+}
+```
+
+Group header view will extend `TableColumnTextGroupHeaderViewBase` and use its template.
+
+### States
+
+N/A
+
+### Accessibility
+
+Accessibility of the cells rendered using the `nimble-table-column-file-size` are handled via the [`nimble-table-cell`](https://github.com/ni/nimble/blob/f663c38741e731bef91aa58e8fb2d1cec653b679/packages/nimble-components/src/table/components/cell/template.ts#L6) which has a `role` of [`cell`](https://w3c.github.io/aria/#cell).
+
+### Mobile
+
+N/A
+
+### Globalization
+
+#### Localization of the number
+
+For this we will use `Intl.NumberFormat`, passing in the value of the `lang` design token as the locale.
+
+#### Localization of the unit label
+
+This is more difficult, as there does not seem to be any API for getting a file size unit string for a given locale. Options:
+
+1. Find translations of each of our unit labels ("byte", "bytes", "KB", "MB", "GB", "TB", "PB", "KiB", "MiB", "GiB", "TiB", "PiB") for a fixed set of languages we wish to support. Maintain a mapping of language codes (e.g. "fr", "de", "zh_CN") to arrays of those localized unit labels. Given a locale to use, look up its language subtag in our map. If not found, fall back to English.
+
+**Pros:** No work for clients.
+
+**Cons:** Only supports a fixed set of languages. Up to us to find accurate translations. Unsure if subtags other than the language (e.g. region) could be relevant to the translation.
+
+2. Use Nimble label provider. Add label tokens for the units, and rely on clients to provide translations.
+
+**Pros:** Supports any language a client cares to provide translations for.
+
+**Cons:** Less convenient for clients. Does not honor the `lang` setting on the page or on `nimble-theme-provider`.
+
+I suggest we go with option 1, primarily because I am hesitant to add twelve new label provider strings that clients are expected to localize. I suggest initially supporting the following languages:
```

Comment:
Given those two options, I think option 2 is better. I don't think maintaining a map of localized strings within nimble is a good idea. I'm not sure how you came up with the list of locales we should initially support, but I don't think it should be the responsibility of nimble to add explicit support for any language that a client wants to use the product in.

We've established the pattern of using the label provider, and I don't think we should avoid using we think it is inconvenient for clients.

Reply:
I agree that Nimble shouldn't pick a subset of languages to support and I agree that Nimble shouldn't do the work to translate every language, so between those options I'd also vote for the label provider approach.

If we say using the Intl.NumberFormat translations (https://github.com/ni/nimble/pull/1566/files#r1338793620) is option 3 then one more idea for option 4 is to pull in a library that does the translation for us. It'd have to be better than the browser in some way, standards-based, and not too big or slow, so I dunno if we'd find one that meets our needs. But one that came up in my searching was [format.js](https://formatjs.io/). I'm not sure if it has any capabilities beyond Intl.NumberFormat though.

Reply:
format.js [uses `Intl.NumberFormat`](https://formatjs.io/docs/intl#formatnumber), so that doesn't provide anything different. I could not find any other libraries that can do the translation for us. I did add two new options that are based on using `Intl.NumberFormat`. I still think option 2 is preferable, though. I'm waiting to hear others' opinions.